fabrics.c

```diff
@@ -505,6 +505,31 @@ static int discover_from_conf_file(nvme_root_t r, nvme_host_t h,
 		errno = 0;
 		ret = nvmf_add_ctrl(h, c, &cfg);
 		if (!ret) {
+			if (persistent) {
+				struct nvme_id_ctrl id = { 0 };
+				ret = nvme_ctrl_identify(c, &id);
+				if (ret)
+					goto next;
+				/* Recreate the controller using the
+				 * unique subsystem nqn, if supported
+				 */
+				if (strcmp(id.subnqn, NVME_DISC_SUBSYS_NAME)) {
+					subsysnqn = id.subnqn;
+					nvme_disconnect_ctrl(c);
+					c = nvme_create_ctrl(r, subsysnqn,
+							transport, traddr,
+							cfg.host_traddr,
+							cfg.host_iface,
+							trsvcid);
+					if (!c)
+						goto next;
+					nvme_ctrl_set_discovery_ctrl(c, true);
+					errno = 0;
+					ret = nvmf_add_ctrl(h, c, &cfg);
+					if (ret)
+						goto next;
+				}
+			}
```

Comment:
I think this should go into a function. The code below looks almost identically.

Reply:
> I think this should go into a function. The code below looks almost identically.

You mean a common helper function? 

I did think about that initially, but then one would have to pass all the nvme_create_ctrl() & nvmf_add_ctrl() params to that helper function, which would make it look quite ugly and unwieldy. So dropped that idea...

Reply:
I don't. Having almost identically code but does the same thing is a nightmare to maintain.